torch/csrc/CudaIPCTypes.cpp

```diff
@@ -0,0 +1,156 @@
+#ifdef USE_CUDA
+#include <torch/csrc/CudaIPCTypes.h>
+#include <map>
+#include <mutex>
+
+namespace torch {
+
+void BadIPCPatternError() {
+  static bool warned = false;
+  if (not warned) {
+    LOG(WARNING)
+        << "Producer process has been terminated before all shared CUDA tensors released.";
+    warned = true;
+  }
+}
+
+struct CudaIPCGlobalEntities {
+  std::mutex mutex;
+  std::map<std::string, std::shared_ptr<CudaIPCRefCountersFile>>
+      ref_counters_files_;
+  std::shared_ptr<CudaIPCRefCountersFile> next_available_ref_counters_file_;
+  CudaIPCSentDataLimbo CudaIPCSentDataLimbo_;
+  CudaIPCGlobalEntities() : ref_counters_files_() {}
```

Comment:
You don't have to do anything about it, but you don't need the explicit initialization of `ref_counters_files_`; as `std::map` is not POD it will be automatically default constructed.

Reply:
I wish. But linter failed here and I had to add it.